testapi.pm

```diff
@@ -475,6 +491,43 @@ sub wait_screen_change(&@) {
 
 ## helpers end
 
+## synchronization API
+use constant {
+    MUTEX_CREATE => 'mutex_create',
+    MUTEX_LOCK   => 'mutex_lock',
+    MUTEX_UNLOCK => 'mutex_unlock',
+};
+
+sub mutex_lock($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_LOCK, $name);
+}
+
+sub mutex_unlock($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_UNLOCK, $name);
+}
+
+sub mutex_create($) {
+    my ($name) = @_;
+    return _mutex_call(MUTEX_CREATE, $name);
+}
+
+sub _mutex_call($$) {
+    my ($action, $name) = @_;
+    # blocking call to worker, worker asks Scheduler for lock
+    bmwqemu::fctlog("Trying mutex action $action on lock $name");
+    my $port = $bmwqemu::vars{'WORKER_PORT'};
+    my $path = "$action/$name";
+    while (1) {
+        my $res = qx(curl -s -o /dev/null -w \"\%{http_code}\" -I http://localhost:$port/$path);
```

Comment:
What do you mean by 'from worker to it'? To os-autoinst http? I can't imagine yet how that is supposed to work for locking. This part is for direction os-autoinst -> worker (worker then acts as a proxy to scheduler).